datafusion/common/src/tree_node.rs

```diff
@@ -18,11 +18,24 @@
 //! This module provides common traits for visiting or rewriting tree
 //! data structures easily.
 
-use std::borrow::Cow;
 use std::sync::Arc;
 
 use crate::Result;
 
+#[macro_export]
+macro_rules! handle_tree_recursion {
```

Comment:
Perhaps we can document what this macro does (aka `return`s from the function if Skip or Stop)